extensions/ql-vscode/src/vscode-tests/factories/remote-queries/shared/variant-analysis.ts

```diff
@@ -10,11 +10,17 @@ import { createMockScannedRepos } from './scanned-repositories';
 import { createMockSkippedRepos } from './skipped-repositories';
 import { createMockRepository } from './repository';
 
-export function createMockVariantAnalysis(
-  status: VariantAnalysisStatus = VariantAnalysisStatus.InProgress,
-  scannedRepos: VariantAnalysisScannedRepository[] = createMockScannedRepos(),
-  skippedRepos: VariantAnalysisSkippedRepositories = createMockSkippedRepos()
-): VariantAnalysis {
+export function createMockVariantAnalysis({
+  status = VariantAnalysisStatus.InProgress,
+  scannedRepos = createMockScannedRepos(),
+  skippedRepos = createMockSkippedRepos(),
+  executionStartTime = faker.datatype.number()
+}: {
+  status?: VariantAnalysisStatus,
+  scannedRepos?: VariantAnalysisScannedRepository[],
+  skippedRepos?: VariantAnalysisSkippedRepositories,
+  executionStartTime?: number | undefined
+}): VariantAnalysis {
```

Comment:
Thanks for making this change, this makes it much easier to use!

An alternative to this would be to allow passing in all properties using something like this:

```ts
export function createMockVariantAnalysis(data: Partial<VariantAnalysis>): VariantAnalysis {
  return {
    status: VariantAnalysisStatus.InProgress,
    // ... other defaults
    ...data,
  };
}
```

What do you think?

Reply:
Nice! I gave it a shot but I think it would involve declaring these params:
```
scannedRepos = createMockScannedRepos(),
skippedRepos = createMockSkippedRepos(),
```
outside of the method call and passing them in. That would require a lot more refactoring and I'm a bit refactored out. 😄 

I considered adding it any way:
```
export function createMockVariantAnalysis({
  status = VariantAnalysisStatus.InProgress,
  scannedRepos = createMockScannedRepos(),
  skippedRepos = createMockSkippedRepos(),
  executionStartTime = faker.datatype.number(),
  data,
}: {
  status?: VariantAnalysisStatus,
  scannedRepos?: VariantAnalysisScannedRepository[],
  skippedRepos?: VariantAnalysisSkippedRepositories,
  executionStartTime?: number | undefined,
  data?: Partial<VariantAnalysis>
}): VariantAnalysis {
  return {
    // all the existing stuff
    ...data
  }
}
```

but that seems like it would defeat the point of it.

Reply:
Ok I've thought this a bit more. I'll try in my next PR to see if we could adapt it.